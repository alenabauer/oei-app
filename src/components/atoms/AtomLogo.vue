<template>
  <div class="atom_logo">
    <img alt="OEI logo" class="atom_logo__image" src="../../assets/logo.png" width="75" height="75" />
    <div class="atom_logo__text">
      <span>Orbital</span>
      <span>Edge</span>
      <span>Imaging</span>
    </div>
  </div>
</template>
<script>
export default {
  name: 'AtomLogo',
}
</script>
<style lang="scss" scoped>
.atom_logo {
  display: flex;
  align-items: center;
  justify-content: start;
  height: fit-content;

  &__image {
    padding: 0.5rem;
    height: 100%;
  }

  &__text {
    flex: 1;
    display: flex;
    flex-direction: column;
    justify-content: center;
    height: 100%;
    background-color: transparent;

    span {
      font-weight: 600;
      line-height: 1.1rem;
      color: var(--color-red);
    }
  }
}
</style>