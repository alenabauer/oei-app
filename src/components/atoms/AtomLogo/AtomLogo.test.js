import AtomLogo from './AtomLogo.vue';
import { mount } from "@vue/test-utils";

it('should render correctly', () => {
    const wrapper = mount(AtomLogo);
    expect(wrapper.html()).toMatchSnapshot();
})

it('should render an image', () => {
    const wrapper = mount(AtomLogo);
    expect(wrapper.html()).toContain('img');
})

it('should render an image with the correct src', () => {
    const wrapper = mount(AtomLogo);
    expect(wrapper.html()).toContain('src="/src/assets/logo.png"');
})