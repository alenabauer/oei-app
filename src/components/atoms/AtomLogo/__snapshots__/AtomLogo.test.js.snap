// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render correctly 1`] = `
"<div data-v-d9e83458=\\"\\" class=\\"atom_logo\\"><img data-v-d9e83458=\\"\\" alt=\\"OEI logo\\" class=\\"atom_logo__image\\" src=\\"/src/assets/logo.png\\" width=\\"75\\" height=\\"75\\">
  <div data-v-d9e83458=\\"\\" class=\\"atom_logo__text\\"><span data-v-d9e83458=\\"\\">Orbital</span><span data-v-d9e83458=\\"\\">Edge</span><span data-v-d9e83458=\\"\\">Imaging</span></div>
</div>"
`;
